### ISSUE TITLE
Gradient clipping is a no-op: `clip_grad_norm_` runs before `backward()` in 9 scripts

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#11]
Every current training loop calls `clip_grad_norm_` **before** `backward()`, so it clips gradients that were just zeroed — the clip is a no-op and the optimizer steps on unclipped gradients.

```python
# scripts/modeling/hits_encdec/train.py:70-74
optimizer.zero_grad(set_to_none=True)
if config["clip_gradients"]:
    torch.nn.utils.clip_grad_norm_(model.parameters(), 0.5)   # grads are None/zero here
loss.backward()
optimizer.step()
```

Verified:

```
[clip-before-backward] norm seen by clip_grad_norm_      = 0.0000
[clip-before-backward] actual grad norm at optimizer.step() = 1510842.0000  (unclipped)
```

Affected (9 sites):

- `scripts/modeling/hits_encdec/train.py:72`
- `scripts/modeling/encdec/train.py:72`
- `scripts/modeling/chroma/train.py:72`
- `scripts/modeling/hits_encdec_polydrums/train.py:520`
- `scripts/modeling/hits_dec/train.py:236`
- `scripts/modeling/hits_dec_xformer/train.py:220`
- `scripts/modeling/rsp_fc/train.py:203`
- `scripts/modeling/tvae/train.py:296` — **unconditional**, so tvae is actually affected today
- `scripts/modeling/tvae/overfit_batch_tvae.py:244` — unconditional

The three older `part_*vae` scripts get the order right, which is presumably where this was copied from before being rearranged.

All the newer configs happen to set `clip_gradients: False`, so for those this is currently only misleading — but tvae clips unconditionally and gets nothing.

**Fix**: `zero_grad()` → `backward()` → `clip_grad_norm_()` → `step()`. Worth pulling the whole backprop block into one shared helper so it can't drift again.
